### PR TITLE
create rdg7 sample

### PR DESCRIPTION
**Pull request by timdeschryver**
Part of https://github.com/dotnet/AspNetCore.Docs/issues/30479
Needs to be merged before https://github.com/dotnet/AspNetCore.Docs/pull/30527

### CODE REVIEW

File: fundamentals/aot/diagnostics/Rdg7/Program.cs [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    
+    private TodoRequest()
+    {
+    }
+}
+
+public record Todo(int Id, string Task);
+
+[JsonSerializable(typeof(Todo[]))]
+internal partial class AppJsonSerializerContext : JsonSerializerContext
```
Comment (timdeschryver) [inline fundamentals/aot/diagnostics/Rdg7/Program.cs:35]: I based myself on other examples, but I think this isn't required?

Comment (Rick-Anderson) [reply fundamentals/aot/diagnostics/Rdg7/Program.cs:35]: It was needed in the first sample and I just copied it. Maybe I can get rid of it in all my samples.

**Comment by Rick-Anderson** [comment]
@timdeschryver  great work. Your changes will be live in 15 minutes. Can you take a look at the live version and open any need issues?

**Comment by timdeschryver** [comment]
@Rick-Anderson thank you. 

The changes are looking good, but I messed up a highlight. I will fix this in an upcoming PR.

https://learn.microsoft.com/en-us/aspnet/core/fundamentals/aot/request-delegate-generator/diagnostics/rdg007?view=aspnetcore-8.0
image

**Comment by Rick-Anderson** [comment]
Easy to fix from the browser.

**Pull request merged**